src/snippetor-CreateFoldersModal.ts

```diff
@@ -0,0 +1,535 @@
+        );
+        colorGroup.createEl("label", {
+            text: label,
+            attr: { for: name },
+        });
+        const taskColor = this.helper.createColorPickerComponent(
+            colorGroup,
+            element,
+            name,
+            update,
+            COLOR.FOREGROUND
+        );
+        // sync light/dark mode
+        this.helper.createColorSyncComponent(
+            colorGroup,
+            element,
+            COLOR.FOREGROUND,
+            (value) => {
+                taskColor.value = this.helper.getPickerValue(
+                    element,
+                    COLOR.FOREGROUND
+                );
+                update(value);
+            }
+        );
+        // reset input element
+        const resetFg = this.helper.createResetColorComponent(
+            colorGroup,
+            async () => {
+                this.helper.clearModeColor(element, COLOR.FOREGROUND);
+                taskColor.value = this.helper.getPickerValue(
+                    element,
+                    COLOR.FOREGROUND
+                );
+                update(undefined);
+            },
+            COLOR.FOREGROUND
+        );
+        resetFg.extraSettingsEl.addClass("no-padding");
+    }
+
+    backgroundColorPicker(
+        container: HTMLElement,
+        element: ColoredElement,
+        name: string,
+        label: string,
+        update: (value: string) => void
+    ) {
+        const colorGroup = container.createSpan(
+            "snippetor-group decorated color"
+        );
+        colorGroup.createEl("label", {
+            text: label,
+            attr: { for: name },
+        });
+        const taskColor = this.helper.createColorPickerComponent(
+            colorGroup,
+            element,
+            name,
+            update,
+            COLOR.BACKGROUND
+        );
+        // sync light/dark mode
+        this.helper.createColorSyncComponent(
+            colorGroup,
+            element,
+            COLOR.BACKGROUND,
+            (value) => {
+                taskColor.value = this.helper.getPickerValue(
+                    element,
+                    COLOR.BACKGROUND
+                );
+                update(value);
+            }
+        );
+        // reset input element
+        const reset = this.helper.createResetColorComponent(
+            colorGroup,
+            async () => {
+                this.helper.clearModeColor(element, COLOR.BACKGROUND);
+                taskColor.value = this.helper.getPickerValue(
+                    element,
+                    COLOR.BACKGROUND
+                );
+                update(undefined);
+            },
+            COLOR.BACKGROUND
+        );
+        reset.extraSettingsEl.addClass("no-padding");
+    }
+
+    verifyDataValue(input: HTMLInputElement) {
+        input.removeClass("data-value-error");
+        input.removeAttribute("aria-label");
+
+        /*         const count = this.cfg.taskSettings.filter(
```

Comment:
Just delete this. It doesn't apply to folders. The task still has it if you need to remember what it does.